etna/models/holt_winters.py

```diff
@@ -277,6 +277,126 @@ def get_model(self) -> HoltWintersResultsWrapper:
         """
         return self._result
 
+    def _check_mul_components(self):
+        """Raise error if model has multiplicative components."""
+        model = self._model
+
+        if model is None:
+            raise ValueError("This model is not fitted!")
+
+        if (model.trend is not None and model.trend == "mul") or (
+            model.seasonal is not None and model.seasonal == "mul"
+        ):
+            raise ValueError("Forecast decomposition is only supported for additive components!")
+
+    def _rescale_components(self, df: pd.DataFrame, components: pd.DataFrame) -> pd.DataFrame:
+        """Rescale components when Box-Cox transform used."""
+        pred = np.sum(components.values, axis=1)
+        transformed_pred = self.predict(df=df)
+        components *= (transformed_pred / pred).reshape((-1, 1))
+        return components
+
+    def forecast_components(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Estimate forecast components.
+
+        Parameters
+        ----------
+        df:
+            features dataframe
+
+        Returns
+        -------
+        :
+            dataframe with forecast components
+        """
+        model = self._model
+        fit_result = self._result
+
+        if fit_result is None or model is None:
+            raise ValueError("This model is not fitted!")
+
+        self._check_mul_components()
+        self._check_df(df)
+
+        horizon = df["timestamp"].nunique()
+        horizon_steps = np.arange(1, horizon + 1)
+
+        components = {"target_component_level": fit_result.level[-1] * np.ones(horizon)}
+
+        if model.trend is not None:
+            t = horizon_steps.copy()
+
+            if model.damped_trend:
+                t = np.cumsum(fit_result.params["damping_trend"] ** t)
```

Comment:
Let's use the analogue from bottleneck for `cumsum`

Reply:
Seems like the bottleneck doesn't have such function